pkg/performanceprofile/controller/performanceprofile/components/machineconfig/machineconfig.go

```diff
@@ -318,9 +327,51 @@ func getIgnitionConfig(profile *performancev2.PerformanceProfile, pinningMode *a
 		Name:     getSystemdService(clearIRQBalanceBannedCPUs),
 	})
 
+	if ok, ovsSliceName := MoveOvsIntoOwnSlice(); ok {
+		// Create the OVS slice that will lift the cpu restrictions for better kernel networking performance
+		// This is technically not necessary as systemd is smart enough
+		// to create a slice when a unit references it.
+		// However, this allows us to set the resources allocated and the cpu balancing
+
+		ovsCgroupUnit, err := getOvsSliceDefinition(ovsSliceName)
+		if err != nil {
+			return nil, err
+		}
+
+		ovsMode := 0644
+		addContent(ignitionConfig, ovsCgroupUnit, "/etc/systemd/system/"+ovsSliceName, &ovsMode)
+
+		// Configure OVS services to use the newly created slice
+		serviceOvsSlice, err := getOvsSliceUsage(ovsSliceName)
+		if err != nil {
+			return nil, err
+		}
+
+		addContent(ignitionConfig, serviceOvsSlice, "/etc/systemd/system/openvswitch.service.d/"+templateOvsSliceUsageFile, &ovsMode)
+		addContent(ignitionConfig, serviceOvsSlice, "/etc/systemd/system/ovs-vswitchd.service.d/"+templateOvsSliceUsageFile, &ovsMode)
+		addContent(ignitionConfig, serviceOvsSlice, "/etc/systemd/system/ovsdb-server.service.d/"+templateOvsSliceUsageFile, &ovsMode)
+
+		// Tell OVN-K to enable dynamic cpu pinning
+		content, err := getTemplatedOvsFile(assets.Configs, filepath.Join("configs", ovsDynamicPinningTriggerFile), ovsSliceName)
+		if err != nil {
+			return nil, err
+		}
+		addContent(ignitionConfig, content, ovsDynamicPinningTriggerHostFile, &ovsMode)
+	}
+
 	return ignitionConfig, nil
 }
 
+func MoveOvsIntoOwnSlice() (bool, string) {
+	// Make sure this does not interfere with SNO and workload partitioning
+	// where OVS is intentionally still running in reserved only due to
+	// workload partitioning restricting the cpuset for OVN-K pods.
+	//
+	// This will be propagated by the ovkube-node cpu affinity logic
+	// and restrict OVS to reserved only.
+	return true, ovsSliceName
```

Comment:
why do we need a function to essentially return two constants?

Reply:
We don't, but I wanted to have a clean code "hook point" for workload partitioning logic and the related comment. It is an artifact that I decided to keep for future development, because it allows nice grouping of the code for this feature.

Reply:
having a "hook point" is no bother